src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaConsumerContext.cs

```diff
@@ -15,84 +15,66 @@ public class KafkaConsumerContext<TKey, TValue> :
         ConsumerContext<TKey, TValue>
         where TValue : class
     {
-        readonly IConsumer<TKey, TValue> _consumer;
-        readonly IConsumerLockContext<TKey, TValue> _lockContext;
-        ChannelExecutor _executor;
+        readonly IConsumerLockContext _lockContext;
+        readonly ChannelExecutorPool<Confluent.Kafka.Partition> _executorPool;
+        readonly Func<Action<IConsumer<byte[], byte[]>, Error>, IConsumer<byte[], byte[]>> _consumerBuilder;
 
         public KafkaConsumerContext(IHostConfiguration hostConfiguration, ReceiveSettings receiveSettings, IHeadersDeserializer headersDeserializer,
-            ConsumerBuilder<TKey, TValue> consumerBuilder, CancellationToken cancellationToken)
+            IDeserializer<TKey> keyDeserializer, IDeserializer<TValue> valueDeserializer,
+            Func<ConsumerBuilder<byte[], byte[]>> consumerBuilderFactory,
+            CancellationToken cancellationToken)
             : base(cancellationToken)
         {
-            var lockContext = new ConsumerLockContext<TKey, TValue>(hostConfiguration, receiveSettings);
-
-            _consumer = consumerBuilder
+            _executorPool = new ChannelExecutorPool<Confluent.Kafka.Partition>(receiveSettings.PrefetchCount, receiveSettings.ConcurrencyLimit);
```

Comment:
let's just use an `int`, since part.Value is an int, and it's just a bunch of fancy shit on top of an int to make it a "value object".

Reply:
We'll use TopicPartition instead here, it will allow handling wildcard